mutacc/builds/build_dataset.py

```diff
@@ -0,0 +1,166 @@
+
+        #orgainize the background files ackording their pedigree
+        self.backgrounds = organize_samples(backgrounds)
+
+    def merge_files(self, out_dir, threads = None):
+        """
+            Merge the background files with the bam files containing
+            the variants.
+
+            Args:
+                out_dir (str): out_dir (str): path to directory where merged
+                bam file is written
+                threads (int or None): Number of threads to use for
+                samtools merge
+        """
+        #list to hold the paths of the synthetic datasets
+        synthetics = []
+        out_dir = parse_path(out_dir, file_type = "dir")
+
+        #for father, mother, and child key in self.backgrounds
+        for key in self.backgrounds.keys():
+            # store the command in merge_cmd
+            merge_cmd = ["samtools", "merge"]
+
+            if threads:
+                merge_cmd.append("-@")
+                merge_cmd.append(str(threads))
+
+            file_name = parse_path(self.backgrounds[key]["bam_file"]).name
+            out_path = out_dir.joinpath("synthetic_"+key+"_"+file_name)
+
+            #Append path to output file for samtools merge
+            merge_cmd.append(str(out_path))
+            #Append path to background bam
+            merge_cmd.append(self.backgrounds[key]["excluded_bam"])
+
+            #Find the bam files in the cases holding the variants
+            for case in self.cases:
+                #Organize sample objects ackording to pedigree (father, mother, child)
+                samples = organize_samples(case["samples"])
+                #for the variant bam file with the same key (father, mother, child)
+                #append it to the command samtools merge
+                merge_cmd.append(samples[key]["variant_bam_file"])
+
+            LOG.info("Merging Files")
+            #Make subprocess for command
+            subprocess.call(merge_cmd)
+            #append the path to synthetic dataset to synthetics
+            synthetics.append(out_path)
+
+        #Remove the 'backround' bam files from disk
+        for key in self.backgrounds.keys():
+            try:
+                os.remove(self.backgrounds[key]["excluded_bam"])
+            except FileNotFoundError:
+                LOG.warning("file {} does not exist".format(
+                    self.backgrounds[key]["excluded_bam"])
+                    )
+        #Return list holding paths to synthetic datasets
+        return synthetics
+
+
+def organize_samples(samples):
+    """
+        Organizes list of samples in a correct pedigree structure.
+    """
+    if len(samples) == 1:
+        return samples
+
+    sample_ids = set(sample["sample_id"] for sample in samples)
+
+    #Find child
+    for sample in samples:
+
+        if sample["mother"] in sample_ids \
+        and sample["father"] in sample_ids:
+
+            child = sample
+            break
+    try:
+        child
```

Comment:
What about siblings? How would these be handled? @moonso Do you have any functions in genmod or other packages that already perform these identity checks?

Reply:
at the moment this only work for father-mother-child trios. I'm trying to come up with a way that would work for all cases. I thought we could discuss this in our next meeting.

Reply:
Yes I agree. This should be handled already when parsing the ped file. In that way you can store something like `sample.is_child=True`. You can look at github.com/moonso/ped_parser if you like.